public/app/features/dashboard/components/SupportSnapshot/SupportSnapshot.tsx

```diff
@@ -71,11 +71,21 @@ export function SupportSnapshot({ panel, plugin, onClose }: Props) {
   const [snapshotText, setDashboardText] = useState('...');
   const [rand, setRand] = useState<Randomize>({});
   const [_, copyToClipboard] = useCopyToClipboard();
+  const [iframeLoading, setIframeLoading] = useState(false);
+  const snapshot = useAsync(async () => {
+    return getDebugDashboard(panel, rand, getTimeSrv().timeRange());
+  }, [rand, panel, plugin]);
+
   const info = useAsync(async () => {
-    const dashboard = await getDebugDashboard(panel, rand, getTimeSrv().timeRange());
-    setDashboardToFetchFromLocalStorage({ meta: {}, dashboard });
-    setDashboardText(JSON.stringify(dashboard, null, 2));
-  }, [rand, panel, plugin, setDashboardText, currentTab]);
+    const dashboard = snapshot.value;
+    if (dashboard) {
+      // When iframe is supported, pass the dashboard via local storage
+      if (iframeLoading && currentTab === InspectTab.Support) {
+        setDashboardToFetchFromLocalStorage({ meta: {}, dashboard });
```

Comment:
@torkelo this avoids issues leaving the stray dashboard in local storage because it is only added set after we know the iframe is loading properly

Reply:
ok, I see. Could be fragile (race condition on what happens first this effect triggering or new route opening inside the iframe).

Reply:
but seems to work